src/backend/distributed/utils/maintenanced.c

```diff
@@ -238,7 +247,6 @@ CitusMaintenanceDaemonMain(Datum main_arg)
 	error_context_stack = &errorCallback;
 
 	/* wire up signals */
-	pqsignal(SIGTERM, die);
```

Comment:
Why did you move this? Until the BackgroundWorkerUnblockSignals signals are blocked...